code/modules/atmospherics/FEA_turf_tile.dm

```diff
@@ -592,50 +591,50 @@ var/global/list/turf/hotly_processed_turfs = list()
 	var/turf/simulated/east = get_step(src,EAST)
 	var/turf/simulated/west = get_step(src,WEST)
 
-	if(need_rebuild)
-		if(istype(center)) //Rebuild/update nearby group geometry
+	if(need_rebuild) // time to make new groups
+		if(center.turf_flags & IS_TYPE_SIMULATED) //Rebuild/update nearby group geometry
 			if(center.parent)
 				air_master.groups_to_rebuild |= center.parent
 			else
 				air_master.tiles_to_update |= src
 
-		if(istype(north))
+		if(north.turf_flags & IS_TYPE_SIMULATED)
 			north.tilenotify(src)
 			if(north.parent)
 				air_master.groups_to_rebuild |= north.parent
 			else
 				air_master.tiles_to_update |= north
-		if(istype(south))
+		if(south.turf_flags & IS_TYPE_SIMULATED)
 			south.tilenotify(src)
 			if(south.parent)
 				air_master.groups_to_rebuild |= south.parent
 			else
 				air_master.tiles_to_update |= south
-		if(istype(east))
+		if(east.turf_flags & IS_TYPE_SIMULATED)
 			east.tilenotify(src)
 			if(east.parent)
 				air_master.groups_to_rebuild |= east.parent
 			else
 				air_master.tiles_to_update |= east
-		if(istype(west))
+		if(west.turf_flags & IS_TYPE_SIMULATED)
```

Comment:
Hm. What does get_step return at the edge of the map? If it returns null, need to use `?.` on those var accesses or this would runtime in that edge case